File: api/Medico.Api/DB/Models/ChiefComplaintTemplate.cs
using System;

namespace Medico.Api.DB.Models
{
    public class ChiefComplaintTemplate
    {
        public Guid ChiefComplaintId { get; set; }

        public ChiefComplaint ChiefComplaint { get; set; }

        public Template Template { get; set; }

        public Guid TemplateId { get; set; }

        public bool IsDelete { get; set; }
    }
}

File: client/src/dataModels/chiefComplaintTemplate.ts
import { ConvertibleFromEntityModel } from "./baseEntityModel";

export class ChiefComplaintTemplate implements ConvertibleFromEntityModel {
    ChiefComplaintId: string;
    IsDelete: boolean;
    TemplateId: string;

    constructor(isDelete: boolean = false) {
        this.ChiefComplaintId = "";
        this.TemplateId = "";
        this.IsDelete = isDelete;
    }

    createFromEntityModel(entity: any) {
        const chiefComplaintTemplate = new ChiefComplaintTemplate(!!entity.isDelete);

        chiefComplaintTemplate.ChiefComplaintId = entity.ChiefComplaintId;
        chiefComplaintTemplate.TemplateId = entity.TemplateId;

        return chiefComplaintTemplate;
    }
}